app/src/main/java/com/example/plantsapp/presentation/ui/authentication/AuthFragment.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.plantsapp.presentation.ui.authentication
+
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.commit
+import androidx.fragment.app.viewModels
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.example.plantsapp.R
+import com.example.plantsapp.databinding.FragmentAuthBinding
+import com.example.plantsapp.presentation.ui.tasksfordays.TasksForDaysFragment
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class AuthFragment : Fragment(R.layout.fragment_auth) {
+
+    private val binding: FragmentAuthBinding by viewBinding(FragmentAuthBinding::bind)
+    private val viewModel: AuthViewModel by viewModels()
+    @Inject
+    lateinit var gso: GoogleSignInOptions
+    private val googleSignInClient by lazy {
+                GoogleSignIn.getClient(requireActivity(), gso)
```

Comment:
Is activity necessary here? 
Maybe you can getClient with application context? And that could help you to hide this inside of Hilt? 

But if not - leave it like this